aws-storage-s3/src/androidTest/java/com/amplifyframework/storage/s3/AWSS3StoragePathGetUrlTest.kt

```diff
@@ -79,4 +82,25 @@ class AWSS3StoragePathGetUrlTest {
         assertEquals("/public/$SMALL_FILE_NAME", result.url.path)
         assertTrue(result.url.query.contains("X-Amz-Expires=30"))
     }
+
+    @Test
+    fun testGetUrlWithObjectExistenceValidationEnabled() {
+        val result = synchronousStorage.getUrl(
+            SMALL_FILE_PATH,
+            AWSS3StorageGetPresignedUrlOptions.builder().setValidateObjectExistence(true).expires(30).build()
+        )
+
+        assertEquals("/public/$SMALL_FILE_NAME", result.url.path)
+        assertTrue(result.url.query.contains("X-Amz-Expires=30"))
+    }
+
+    @Test
+    fun testGetUrlWithStorageExceptionObjectNotFoundThrown() {
+        assertThrows(StorageException::class.java) {
+            synchronousStorage.getUrl(
+                StoragePath.fromString("public/SOME_UNKNOWN_FILE"),
+                AWSS3StorageGetPresignedUrlOptions.builder().setValidateObjectExistence(true).expires(30).build()
+            )
+        }
```

Comment:
can we assert to make sure this exception is the excepted notFound exception? In current commit, that should be checking inner exception i believe

Reply:
integration test updated to specifically check for `NotFound` exception